lib/modules/versioning/rpm/index.ts

```diff
@@ -0,0 +1,283 @@
+    // 3.12.0-1~^20231001 LOSES to 3.12.0-1^20231001
+    const snapshotIndex = remainingVersion.indexOf('^');
+
+    if (releaseIndex >= 0) {
+      upstreamVersion = remainingVersion.slice(0, releaseIndex);
+
+      // Do NOT splice out prerelease, we need to distinguish if the flag is set or not, regardless if there is a version.
+      // The tilde will get filtered out during regex
+      if (prereleaseIndex >= 0) {
+        rpmRelease = remainingVersion.slice(releaseIndex, prereleaseIndex);
+        if (snapshotIndex >= 0) {
+          rpmPreRelease = remainingVersion.slice(
+            prereleaseIndex,
+            snapshotIndex,
+          );
+          snapshot = remainingVersion.slice(snapshotIndex + 1);
+        } else {
+          rpmPreRelease = remainingVersion.slice(prereleaseIndex);
+        }
+      } else {
+        rpmRelease = remainingVersion.slice(releaseIndex + 1);
+      }
+    } else {
+      upstreamVersion = remainingVersion;
+    }
+
+    upstreamVersion;
+
+    const release = [...remainingVersion.matchAll(regEx(/\d+/g))].map((m) =>
+      parseInt(m[0], 10),
+    );
+
+    return {
+      epoch,
+      upstreamVersion,
+      rpmRelease,
+      release,
+      rpmPreRelease,
+      snapshot,
+    };
+  }
+
+  protected _compare_string(s1: string, s2: string): number {
+    if (s1 === s2) {
+      return 0;
+    }
+
+    const minLength = Math.min(s1.length, s2.length);
+
+    for (let i = 0; i < minLength; i++) {
+      const c1 = s1[i];
+      const c2 = s2[i];
+
+      if (c1 === c2) {
+        continue;
+      }
+
+      if (c1 > c2) {
+        return 1;
+      } else if (c1 < c2) {
+        return -1;
+      }
+    }
+
+    // Okay, they've been the exact same up until now, so return the longer one
+    return s1.length > s2.length ? 1 : -1;
+  }
+
+  /**
+   * Taken from https://github.com/rpm-software-management/rpm/blob/master/rpmio/rpmvercmp.c
+   */
+  protected _compare_glob(v1: string, v2: string): number {
+    if (v1 === v2) {
+      return 0;
+    }
+
+    const matchesv1 = v1.match(alphaNumPattern) ?? [];
+    const matchesv2 = v2.match(alphaNumPattern) ?? [];
+    const matches = Math.min(matchesv1.length, matchesv2.length);
+
+    for (let i = 0; i < matches; i++) {
+      const matchv1 = matchesv1[i];
+      const matchv2 = matchesv2[i];
+
+      // compare tildes
+      if (matchv1?.[0] === '~' || matchv2?.[0] === '~') {
+        if (matchv1?.[0] !== '~') {
+          return 1;
+        }
+
+        if (matchv2?.[0] !== '~') {
+          return -1;
+        }
+      }
+
+      if (is.numericString(matchv1?.[0])) {
+        // numbers are greater than letters
+        if (!is.numericString(matchv2?.[0])) {
+          return 1;
+        }
+
+        //We clearly have a number here, so return which is greater
+        const num1 = Number.parseInt(matchv1, 10);
+        const num2 = Number.parseInt(matchv2, 10);
+
+        const result = num1 - num2;
```

Comment:
```suggestion
        const result = matchv1.localCompare(matchv2, undefined, { numeric: true });
```
isn't this the same result but much faster?

Reply:
I'm not the best with javascript, so you're probably right haha.